pandas/core/indexes/multi.py

```diff
@@ -2058,7 +2058,7 @@ def drop(self, codes, level=None, errors="raise"):
 
         if not isinstance(codes, (np.ndarray, Index)):
             try:
-                codes = com.index_labels_to_array(codes)
+                codes = com.index_labels_to_array(codes, dtype=object)
```

Comment:
FYI, at least in our tests, this isn't changing behavior. I added an `assert com.index_labels_to_array(codes, dtype=object).dtype == com.index_labels_to_array(codes)` temporarily,  so we were always inferring object dtype here (in our tests).